packages/react/src/cookies-utils.tsx

```diff
@@ -0,0 +1,71 @@
+import {parse} from 'worktop/cookie';
+import {SHOPIFY_Y, SHOPIFY_S} from './cart-constants.js';
+
+const zeros = '00000000';
+const tokenHash = 'xxxx-4xxx-xxxx-xxxxxxxxxxxx';
+
+export function buildUUID(): string {
+  let hash = '';
+
+  try {
+    const crypto: Crypto = window.crypto;
+    const randomValuesArray = new Uint16Array(31);
+    crypto.getRandomValues(randomValuesArray);
+
+    // Generate a strong UUID
+    let i = 0;
+    hash = tokenHash
+      .replace(/[x]/g, (c: string): string => {
+        const r = randomValuesArray[i] % 16;
+        const v = c === 'x' ? r : (r & 0x3) | 0x8;
+        i++;
+        return v.toString(16);
+      })
+      .toUpperCase();
+  } catch (err) {
+    // crypto not available, generate weak UUID
+    hash = tokenHash
+      .replace(/[x]/g, (c: string): string => {
+        const r = (Math.random() * 16) | 0;
+        const v = c === 'x' ? r : (r & 0x3) | 0x8;
+        return v.toString(16);
+      })
+      .toUpperCase();
+  }
+
+  return `${hexTime()}-${hash}`;
+}
+
+export function hexTime(): string {
+  // 32 bit representations of new Date().getTime() and performance.now()
+  let dateNumber = 0;
+  let perfNumber = 0;
+
+  // Result of zero-fill right shift is always positive
+  dateNumber = new Date().getTime() >>> 0;
```

Comment:
wut hahaha. I'm going to have to play around and see what this is doing. Apparently something with positive integers... which makes sense in a function called `hexTime`. Hmmm